.github/actions/artifact_definition/action.yaml

```diff
@@ -0,0 +1,77 @@
+name: Artifact Definition
+description: Action to generate and upload an artifact
+
+# This action allows you to generate an artifact in a ZIP file and upload it.
+
+inputs:
+  workspace_name:
+    description: The name of the workspace to create the artifact for.
+    type: string
+    required: true
+  bundle_name:
+    description: Name of the bundle
+    type: string
+    required: false
+  base_workdir:
+    description: The base workdir (Default is ./out)
+    type: string
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      name: Checkout
+
+    - name: Build the artifact
```

Comment:
The idea behind creating multiple actions instead of a specific workflow was to build several reusable modules. However, if we notice too many specificities in the actions, I would suggest considering returning to a single workflow approach, such as [web_app_deploy](https://github.com/pagopa/dx/blob/main/.github/workflows/web_app_deploy.yaml).